### PR TITLE
feature:support configuring apolloService and apolloCluster

### PR DESCRIPTION
**Pull request by slinpq**

### CODE REVIEW

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java [hunk 1]
```diff
@@ -54,17 +54,21 @@
     private static final String APP_ID = "appId";
     private static final String APOLLO_META = "apolloMeta";
     private static final String APOLLO_SECRET = "apolloAccesskeySecret";
+    private static final String APOLLO_CLUSTER = "cluster";
```
Comment (a364176773) [inline config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:57]: APOLLO_CLUSTER = "seata"  Is that better?

File: server/src/main/resources/registry.conf [hunk 2]
```diff
@@ -57,7 +57,7 @@ registry {
 
 config {
   # file、nacos 、apollo、zk、consul、etcd3
-  type = "file"
+  type = "apollo"
```
Comment (a364176773) [inline server/src/main/resources/registry.conf:60]: file

File: server/src/main/resources/registry.conf [hunk 3]
```diff
@@ -71,9 +71,12 @@ config {
   }
   apollo {
     appId = "seata-server"
+    ## apolloConfigService will cover apolloMeta
     apolloMeta = "http://192.168.1.204:8801"
+    apolloConfigService = "http://192.168.1.204:8080"
     namespace = "application"
     apolloAccesskeySecret = ""
+    cluster = "cluster"
```
Comment (a364176773) [inline server/src/main/resources/registry.conf:79]: cluster = "seata"  Is that better?

**Review by a364176773 — CHANGES_REQUESTED** [review]

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java (hunk 1 above)
Comment (slinpq) [reply config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:57]: yes

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java [hunk 4]
```diff
@@ -176,6 +180,12 @@ private void readyApolloConfig() {
                 System.setProperty(PROP_APOLLO_SECRET, secretKey);
             }
         }
+        if (!properties.containsKey(APOLLO_CLUSTER)) {
+            System.setProperty(PROP_APOLLO_CLUSTER, FILE_CONFIG.getConfig(getApolloCluster()));
```
Comment (a364176773) [inline config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:184]: If the value is ""?

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java [hunk 5]
```diff
@@ -176,6 +180,12 @@ private void readyApolloConfig() {
...
+        }
+        if (!properties.containsKey(APOLLO_CONFIG_SERVICE)) {
+            System.setProperty(PROP_APOLLO_CONFIG_SERVICE, FILE_CONFIG.getConfig(getApolloConfigService()));
```
Comment (a364176773) [inline config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:187]: If the value is ""?

**Review by a364176773 — CHANGES_REQUESTED** [review]

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java (hunk 4 above)
Comment (slinpq) [reply config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:184]: It is "default",the default value of apollo.

Comment (slinpq) [reply config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:184]: It is "default",the default value of apollo.

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java (hunk 5 above)
Comment (slinpq) [reply config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:187]: It is empty.Apollo will use apolloMeta.If apolloMeta is empty.The addr of apollo will be null.

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java (hunk 4 above)
Comment (a364176773) [reply config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:184]: > It is "default",the default value of apollo.

I mean, if it's empty, don't set it

Comment (slinpq) [reply config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:184]: The apollo client will set 'default',if it's empty.

Comment (slinpq) [reply config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:184]: The apollo client will set 'default',if it's empty.

File: server/src/main/resources/registry.conf (hunk 3 above)
Comment (a364176773) [reply server/src/main/resources/registry.conf:79]: change

**Review by a364176773 — APPROVED** [review]
LGTM

File: config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java [hunk 6]
```diff
@@ -54,17 +54,21 @@
...
+    private static final String PROP_APOLLO_CLUSTER = "apollo.cluster";
     private static final String NAMESPACE = "namespace";
     private static final String DEFAULT_NAMESPACE = "application";
     private static final Configuration FILE_CONFIG = ConfigurationFactory.CURRENT_FILE_INSTANCE;
     private static volatile Config config;
     private ExecutorService configOperateExecutor;
     private static final int CORE_CONFIG_OPERATE_THREAD = 1;
     private static final ConcurrentMap<String, Set<ConfigurationChangeListener>> LISTENER_SERVICE_MAP
-        = new ConcurrentHashMap<>();
+            = new ConcurrentHashMap<>();
```
Comment (xingfudeshi) [inline config/seata-config-apollo/src/main/java/io/seata/config/apollo/ApolloConfiguration.java:71]: Pls Give it a initial size.

**Review by xingfudeshi — APPROVED** [review]
LGTM.

**Review by xingfudeshi — CHANGES_REQUESTED** [review]
Pls add those config items in yml/properties files

**Comment by a364176773** [comment]
> Pls add those config items in yml/properties files

+1

**Review by xingfudeshi — APPROVED** [review]
LGTM.

**Pull request merged**